contracts/src/factory/facets/architect/ArchitectBase.sol

```diff
@@ -338,3 +346,195 @@ abstract contract ArchitectBase is Factory, IArchitectBase {
+    IRuleEntitlementV2 ruleEntitlementV2 = IRuleEntitlementV2(
+      _deployEntitlement(ims.ruleEntitlement, spaceAddress)
+    );
+
+    if (
+      keccak256(abi.encodePacked(ruleEntitlementV2.moduleType())) !=
+      keccak256(abi.encodePacked("RuleEntitlementV2"))
+    ) {
+      revert Architect__InvalidEntitlementVersion();
+    }
+
+    address[] memory entitlements = new address[](2);
+    entitlements[0] = address(userEntitlement);
+    entitlements[1] = address(ruleEntitlementV2);
+
+    // set entitlements as immutable
+    IEntitlementsManager(spaceAddress).addImmutableEntitlements(entitlements);
+
+    // create minter role with requirements
+    _createMinterEntitlementV2(
+      spaceAddress,
+      userEntitlement,
+      ruleEntitlementV2,
+      spaceInfo.membership.requirements
+    );
+
+    // create member role with membership as the requirement
+    uint256 memberRoleId = _createMemberEntitlement(
+      spaceAddress,
+      spaceInfo.membership.settings.name,
+      spaceInfo.membership.permissions,
+      userEntitlement
+    );
+
+    // create default channel
+    _createDefaultChannel(spaceAddress, memberRoleId, spaceInfo.channel);
+
+    // transfer nft to sender
+    IERC721A(address(ims.spaceToken)).safeTransferFrom(
+      address(this),
+      msg.sender,
+      spaceTokenId
+    );
+
+    // emit event
+    emit SpaceCreated(msg.sender, spaceTokenId, spaceAddress);
+  }
+
+  function _createMinterEntitlementV2(
+    address spaceAddress,
+    IUserEntitlement userEntitlement,
+    IRuleEntitlementV2 ruleEntitlementV2,
+    MembershipRequirementsV2 memory requirements
+  ) internal returns (uint256 roleId) {
+    string[] memory joinPermissions = new string[](1);
+    joinPermissions[0] = Permissions.JoinSpace;
+
+    roleId = IRoles(spaceAddress).createRole(
+      MINTER_ROLE,
+      joinPermissions,
+      new IRolesBase.CreateEntitlement[](0)
+    );
+
+    if (requirements.everyone) {
+      address[] memory users = new address[](1);
+      users[0] = EVERYONE_ADDRESS;
+
+      IRoles(spaceAddress).addRoleToEntitlement(
+        roleId,
+        IRolesBase.CreateEntitlement({
+          module: userEntitlement,
+          data: abi.encode(users)
+        })
+      );
+    } else {
+      if (requirements.users.length != 0) {
+        // validate users
+        for (uint256 i = 0; i < requirements.users.length; ) {
+          Validator.checkAddress(requirements.users[i]);
+          unchecked {
+            i++;
+          }
+        }
+
+        IRoles(spaceAddress).addRoleToEntitlement(
+          roleId,
+          IRolesBase.CreateEntitlement({
+            module: userEntitlement,
+            data: abi.encode(requirements.users)
+          })
+        );
+      }
+
+      IRoles(spaceAddress).addRoleToEntitlement(
+        roleId,
+        IRolesBase.CreateEntitlement({
+          module: ruleEntitlementV2,
+          data: requirements.ruleDataV2
```

Comment:
minor deviation: no decoding here since ruleDataV2 is bytes, as discussed with @giuseppecrj